portfolio/src/main/webapp/visuals.js

```diff
@@ -0,0 +1,180 @@
+google.charts.load('current', { 'packages': ['corechart'] });
+google.charts.setOnLoadCallback(drawChart);
+
+/** Creates the pie chart and adds it to the page. */
+function drawChart() {
+  fetch('cuisine-data').then(response => response.json())
+    .then((cuisineVotes) => {
+
+    let data = new google.visualization.DataTable();
+    data.addColumn('string', 'Cuisine');
+    data.addColumn('number', 'Votes');
+    Object.keys(cuisineVotes).forEach((cuisine) => {
+      data.addRow([cuisine, cuisineVotes[cuisine]]);
+    });
+
+    const options = {
+      pieHole: 0.4,
+      pieSliceTextStyle: {
+        color: '#f5f6fa'
+      },
+      backgroundColor: '#a6ded2',
+      legend: 'none'
+    };
+
+    const chart = new google.visualization.PieChart(document.getElementById('chart-container'));
+    chart.draw(data, options);
+  })
+}
+
+let map;
+let courtsInfo;
+
+/** Creates a map and adds it to the page. */
+function createMap() {
+  fetch('/basketball-courts').then(response => response.json()).then((courts) => {
+    map = new google.maps.Map(document.getElementById('map-container'), {
+      center: { lat: 40.7128, lng: -74.0060 },
+      zoom: 10,
+      styles: style
+    });
+
+    courtsInfo = courts;
+  });
+}
+
+let courtsMarkers = [];
+
+/** add markers for courts */
+function showCourtsMarkers() {
+  clearCourtsMarkers();
+  for (let i = 0; i < courtsInfo.length; i++) {
+    addMarkerWithTimeout(courtsInfo[i], i * 200);
+  }
+}
+
+/**
+ * Drops the markers so that the user can see them falling.
+ * The pins indicate the location of the courts. You can also
+ * click on them to get information about the name and address.
+ * The circles indicate the number of courts (if the property is given).
+ * Also, the radius of the circle is an indication of the number of courts.
+ */
+function addMarkerWithTimeout(court, timeout) {
+  window.setTimeout(function() {
+    courtsMarkers.push(new google.maps.Marker({
+      position: { lat: parseFloat(court.lat), lng: parseFloat(court.lon) },
+      map: map,
+      animation: google.maps.Animation.DROP,
+      title: court.Name
+    }).addListener('click', function() {
+        new google.maps.InfoWindow({
+          content: '<div style="font-weight:bold;">' + court.Name +
+            '</div><div style="font-style:italic;">' + court.Location + '</div>'
+        }).open(map, this);
+    }));
+    if (court.hasOwnProperty('Num_of_Courts')) {
+      let location = new google.maps.LatLng(parseFloat(court.lat), parseFloat(court.lon));
+      courtsMarkers.push(new google.maps.Circle({
+        strokeColor: '#FF0000',
+        strokeOpacity: 0.8,
+        strokeWeight: 2,
+        fillColor: '#FF0000',
+        fillOpacity: 0.35,
+        map: map,
+        center: location,
+        radius: parseInt(court.Num_of_Courts) * 1000
```

Comment:
why is Num_of_courts not already an int?you should do all your parsing like this when the data comes back from server. everywhere else in your client code you have clean data to work with.